model/jpa/src/main/java/org/keycloak/models/jpa/RealmAdapter.java

```diff
@@ -1245,8 +1249,8 @@ public ClientModel getMasterAdminClient() {
 
     @Override
     public void setMasterAdminClient(ClientModel client) {
-        ClientEntity appEntity = client !=null ? em.getReference(ClientEntity.class, client.getId()) : null;
-        realm.setMasterAdminClient(appEntity);
+        String appEntityId = client !=null ? em.getReference(ClientEntity.class, client.getId()).getId() : null;
+        realm.setMasterAdminClient(appEntityId);
         em.flush();
```

Comment:
```suggestion
        realm.setMasterAdminClient(client.getId());
```

I don't think we need to lookup the client here, nor do we need an em.flush here.